Lib/test/test_urllib.py

```diff
@@ -1463,6 +1463,28 @@ def open_spam(self, url):
                 "spam://c:|windows%/:=&?~#+!$,;'@()*[]|/path/"),
                 "//c:|windows%/:=&?~#+!$,;'@()*[]|/path/")
 
+    def test_urlopener_retrieve_file(self):
+        with self.assertWarns(DeprecationWarning):
```

Comment:
`DeprecationWarning` is already tested elsewhere. It's out of scope of this test. Use the `support.ignore_warnings` decorator to suppress it.